src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java

```diff
@@ -115,4 +132,34 @@ public static String getTextAttributeValue(DetailAST ast) {
         return text;
     }
 
+    /**
+     * Returns xpath query results on file as string.
+     *
+     * @param xpath query to evaluate
+     * @param file file to run on
+     * @return all results as string separated by {@link XpathUtil#DELIMITER}
+     * @throws CheckstyleException if some parsing error happens
+     * @throws IOException if an error occurs
+     */
+    public static String printXpathBranch(String xpath, File file) throws CheckstyleException,
+            IOException {
+        final XPathEvaluator xpathEvaluator = new XPathEvaluator();
+        try {
+            final RootNode rootNode = new RootNode(JavaParser.parseFile(file,
+                JavaParser.Options.WITHOUT_COMMENTS));
```

Comment:
@strkkk , sorry, misplaced my [previous comment](https://github.com/checkstyle/checkstyle/pull/7025/commits/6a44d4f25010f635b50ad9d5508ea117378f2741#r319198876).
I think we need comments in parse tree, as some Check (non-javadoc check) can place violation on comment nodes.

Extra test on xpath to comment node is always welcome.

Reply:
Ok, I changed it and added tests for comment xpath evaluation, please take a look